lib/matplotlib/colors.py

```diff
@@ -1545,11 +1545,11 @@ class LogNorm(Normalize):
 
     def autoscale(self, A):
         # docstring inherited.
-        super().autoscale(np.ma.masked_less_equal(A, 0, copy=False))
+        super().autoscale(np.ma.array(A, mask=(A <= 0)))
```

Comment:
Does this save the copy step? It seems like this is creating a new array anyways, so I'm wondering if it is gaining anything relative to just removing the `copy=False`?

Reply:
The default for `np.ma.array` is `copy=False, keep_mask=True`. The former will not copy the data, and the latter will use the existing mask (if any) with the new mask, which will make a copy. Removing `copy=False` from `masked_less_equal` makes a copy of both.

Reply:
Thanks for clarifying! I was curious how to test if this actually holds and learned about `np.shares_memory` today. I did verify this assertion with that.

Reply:
You can also check, if you have `b = np.ma.array(a)`, that `b.base is a` (if `a` is originally a plain array), or `b.base is a.base` (if `a` is originally a masked array).